plugins/wallabag.koplugin/main.lua

```diff
@@ -329,9 +329,17 @@ end
 -- @treturn int 1 failed, 2 skipped, 3 downloaded
 function Wallabag:download(article)
     local skip_article = false
-    local item_url = "/api/entries/" .. article.id .. "/export.epub"
     local title = util.getSafeFilename(article.title, self.directory, 230, 0)
-    local local_path = self.directory .. article_id_prefix .. article.id .. article_id_postfix .. title .. ".epub"
+    local file_ext = ".epub"
+    local item_url = "/api/entries/" .. article.id .. "/export.epub"
+
+    -- If the article links to a pdf file, we will download it directly
```

Comment:
Btw, does how does Wallabag deal with other supported filetypes? Most notably DjVu, EPUB, FB2 & CBZ. Not necessarily for this PR, but it might be worth considering doing something with http://koreader.rocks/doc/modules/document.documentregistry.html#DocumentRegistry:hasProvider or the like if Wallabag works as a transport.

Reply:
Wallabag-converted EPUB files also get outputted as unreadable mess, it would totally make sense to skip the conversion for all files that have a provider.

I tried using the "hasProvider" family of functions to do what I want, but I was stuck on the fact that all of them use `filepath` as input. It's easy enough to add a dict to track the supported mimetypes, but I also have to find the extension to add to my downloaded file.

That extension would need to be the one associated to the preferred provider, but `getProvider` need the extension... I'll try to tackle that later.

Maybe an acceptable hack for the meantime would be to add some more hardcoded extensions/mimetypes for the types above?

Reply:
Sidenote for future refinement, Wallabag can grab files from websites that require authentication (see [here](https://wallabag.org/en/news/new-release-wallabag-2-3-is-there-really)). I'm not sure if this is something the API mentions so it can immediately be taken into account because I've personally never tried it, but it might be something to keep in mind.